java-checks-test-sources/spring-3.2/src/main/java/checks/SQLInjectionCheckSample.java

```diff
@@ -0,0 +1,60 @@
+package checks;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcOperations;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.PreparedStatementCallback;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.jdbc.core.simple.JdbcClient;
+import org.springframework.security.core.userdetails.jdbc.JdbcDaoImpl;
+import org.springframework.security.provisioning.JdbcUserDetailsManager;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class SQLInjectionCheckSample {
+
+  private final JdbcTemplate jdbcTemplate;
+  private final JdbcOperations jdbcOperations;
+  private final NamedParameterJdbcTemplate namedParameterJdbcTemplate;
+  private final JdbcClient jdbcClient;
+
+  @Autowired
+  public SQLInjectionCheckSample(JdbcTemplate jdbcTemplate,
+    JdbcOperations jdbcOperations,
+    NamedParameterJdbcTemplate namedParameterJdbcTemplate,
+    JdbcClient jdbcClient) {
+    this.jdbcTemplate = jdbcTemplate;
+    this.jdbcOperations = jdbcOperations;
+    this.namedParameterJdbcTemplate = namedParameterJdbcTemplate;
+    this.jdbcClient = jdbcClient;
+  }
```

Comment:
I have no strong opinion about it, but: to me, this just seems to make the test more complicated than necessary (thus, distracting). You want to reproduce the pattern for Spring components here (including dependency injection), which is generally a good idea, but on the other hand, it's not required for the test. The sample methods that need any of these instances could just have them as parameters instead.